### PR TITLE
prepare for upload refactor

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/content_graph/parsers/content_item.py [hunk 1]
```diff
@@ -161,7 +164,7 @@ def is_package(path: Path) -> bool:
 
     @staticmethod
     def is_unified_file(path: Path) -> bool:
-        return path.suffix in UNIFIED_FILES_SUFFIXES
+        return path.suffix in UNIFIED_FILES_SUFFIXES and len(path.parts) > 2 and path.parts[-2] in ContentType.folders()
```
Comment (tkatzir) [inline demisto_sdk/commands/content_graph/parsers/content_item.py:167]: Good catch! :-)

File: demisto_sdk/commands/unify/xdrc_template_unifier.py [hunk 2]
```diff
@@ -1,72 +1,25 @@
...
-        self.input_xdrc_template = input
+class XDRCTemplateUnifier(Unifier):
 
-        if dir_name:
-            self.dir_name = dir_name
+    @staticmethod
+    def unify(path: Path, data: dict, marketplace: MarketplaceVersions = None, **kwargs) -> dict:
+        click.echo(f'Unifying {path}...')
+        data = XDRCTemplateUnifier._insert_yaml_template(path, data)
+        click.secho(f'Successfully created unified JSON in {path}', fg="green")
```
Comment (tkatzir) [inline demisto_sdk/commands/unify/xdrc_template_unifier.py:16]:
```suggestion
        click.secho(f'Successfully created a unified JSON in {path}', fg="green")
```

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -2761,24 +2761,23 @@ def get_scripts_and_commands_from_yml_data(data, file_type):
...
+    copy_dict = content_item.copy()  # for modifying dict while iterating
     for field, value in copy_dict.items():
         if field.endswith('_x2'):
-            current_dict[field[:-3]] = value
-            current_dict.pop(field)
-        elif isinstance(current_dict[field], dict):
-            alternate_item_fields(current_dict[field])
-        elif isinstance(current_dict[field], list):
-            for item in current_dict[field]:
+            content_item[field[:-3]] = value
```
[the remark this answers is not in the document]
Comment (ilaner) [reply demisto_sdk/commands/common/tools.py:2775]: we currently don't have fields with `x3` in there.
I think we should take the X2 names to XPANSE as well

[the remark this answers is not in the document]
Comment (ilaner) [reply demisto_sdk/commands/common/tools.py:2775]: what do you mean?

File: demisto_sdk/commands/unify/integration_script_unifier.py [hunk 4]
```diff
@@ -41,185 +38,126 @@
...
+    def unify(
+        path: Path,
+        data: dict,
+        marketplace: MarketplaceVersions = None,
+        custom: str = '',
+        image_prefix: str = DEFAULT_IMAGE_PREFIX,
+        force: bool = False,
+        **kwargs,
+    ):
+        print(f"Merging package: {path}")
```
Comment (tkatzir) [inline demisto_sdk/commands/unify/integration_script_unifier.py:53]:
```suggestion
        print(f"Unifying package: {path}")
```

File: demisto_sdk/commands/unify/unifier.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+from demisto_sdk.commands.common.constants import MarketplaceVersions
+
+
+class Unifier(ABC):
+    """Interface to YAML objects that need to be unified"""
```
Comment (tkatzir) [inline demisto_sdk/commands/unify/unifier.py:8]:
```suggestion
    """Interface to objects that need to be unified into YAML"""
```

File: demisto_sdk/__main__.py [hunk 6]
```diff
@@ -291,21 +280,16 @@ def unify(**kwargs):
...
-
-        # pass arguments to YML unifier and call the command
-        yml_unifier = IntegrationScriptUnifier(**kwargs, custom=custom)
-        yml_unifier.unify()
-
+        PrepareUploadManager.prepare_for_upload(**kwargs)
     return 0
 
 
+main.add_command(prepare_content, name='unify')
```
[the remark this answers is not in the document]
Comment (ilaner) [reply demisto_sdk/__main__.py:292]: added

File: demisto_sdk/commands/prepare_content/prepare_upload_manager.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+    @staticmethod
+    def prepare_for_upload(
+            input: Path,
+            output: Optional[Path] = None,
+            marketplace: MarketplaceVersions = MarketplaceVersions.XSOAR,
+            force: bool = False,
+            **kwargs) -> Path:
+        # This is to force unifying deprecated content items
+        demisto_sdk.commands.content_graph.parsers.content_item.MARKETPLACE_MIN_VERSION = '0.0.0'
```
Comment (tkatzir) [inline demisto_sdk/commands/prepare_content/prepare_upload_manager.py:20]: Isn't it enough?
```suggestion
        demisto_sdk.commands.content_graph.parsers.content_item.MARKETPLACE_MIN_VERSION = '0'
```

**Review by tkatzir — APPROVED** [review]
Great!
One question.

**Pull request merged**